aptos-move/framework/aptos-framework/sources/TokenV1.move

```diff
@@ -698,6 +702,38 @@ module AptosFramework::TokenV1 {
         }
     }
 
+    public(script) fun burn_script(
+        account: &signer,
+        token_id: TokenId,
+        amount: u64,
+    ) acquires TokenStore, Collections, TokenCapabilityStore {
+        let addr = Signer::address_of(account);
+        assert!(balance_of(addr, token_id) >= amount, EINSUFFICIENT_BALANCE);
+        let token = withdraw_token(account, token_id, amount);
+        burn(account, token)
+    }
+
+    public fun burn(cap_owner: &signer, token: Token) acquires Collections, TokenCapabilityStore {
+        assert!(exist_burn_capability(Signer::address_of(cap_owner), token.id), ENO_BURN_CAPABILITY);
```

Comment:
We really need to talk about what capabilities are.

In security systems, it is not just an English word with some informal meaning. But you are currently using it like this, as if someone "has the capability to burn". 

Technically, a capability is a *value* which is a transient token which witnesses a successful authentication step. This value only lives for the duration of transaction. It should never be stored in memory (and not have the `store` ability). Therefore, whenever you have a capability value in hand, you know that this was obtained via an auth step *in that same transaction*, and not hijacked or stolen in other ways. In turn, all legal auth steps must at some point involve a *signer*, because that is the only way how we can prove identity. 

A burn capability would be acquired via an auth step as follows:

```
let burn_cap = acquire_burn_capability(account);
...

fun acquire_burn_capability(a: &signer): BurnCap {
   assert!(exists<BurnAuthorization>(signer_of(a))); BurnCap{address: signer_of(a)}
}   
```

It might be practically marginal, but I'm suggesting to not call the resource capability, but something else like authorization. The notion of capability should be preserved for the transient things we are passing around.

In your code, you may want to acquire the burn cap in `burn_script` and pass it into the `burn` function instead of `cap_owner` signer.

Reply:
I see. I can call the current TokenCapabilityStore as TokenAuthorizationStore. I could then add a new struct ( eg: burn_capability) as the transient burn capability